03week/ticTacToe.js

```diff
@@ -24,35 +25,111 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  for (var i = 0; i <= 2; i++) {
+    let count = 0;
+    for (var j = 0; j <= 2; j++) {
+      if (board[i][j] === playerTurn) {
+        count++;
+        if (count === 3) {
+          return true;
+        }
+      } else {
+        break;
+      }
+    }
+  }
+  return false;
 }
 
 function verticalWin() {
-  // Your code here
+  for (var i = 0; i <= 2; i++) {
+    let count = 0;
+    for (var j = 0; j <= 2; j++) {
+      if (board[j][i] === playerTurn) {
+        count++;
+        if (count === 3) {
+          return true;
+        }
+      } else {
+        break;
+      }
+    }
+  }
+  return false;
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn &&
+      board[1][1] === playerTurn &&
+      board[2][2] === playerTurn)
+      ||
+      (board[0][2] === playerTurn &&
+      board[1][1] === playerTurn &&
+      board[2][0] === playerTurn)) {
+        return true;
+      } else {
+        return false;
+      }
 }
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin() || verticalWin() || diagonalWin()) {
+    console.log(`Player ${playerTurn} won!`);
+    return true;
+  } else {
+    return false;
+  }
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+  board[row][column] = playerTurn;
+  movesCounter++;
+
+  if (checkForWin()) {
+    return false;
+  }
+
+  if (movesCounter === 9) {
+    console.log("It's a tie!");
+    return false;
+  }
+
+  if (playerTurn === "X") {
+    playerTurn = "O";
+  } else {
+    playerTurn = "X";
+  }
+  return true;
+}
+
+function validate(row, column) {
+  if (isNaN(row) || isNaN(column) || row === '' || column === '') {
+    return false;
+  }
+
+  if (board[row][column] === ' ') {
+    return true;
+  } else {
+    return false;
+  }
 }
 
 function getPrompt() {
   printBoard();
   console.log("It's Player " + playerTurn + "'s turn.");
   rl.question('row: ', (row) => {
     rl.question('column: ', (column) => {
-      ticTacToe(row, column);
+      if (validate(row, column)) {
```

Comment:
all of this logic should go in the ticTacToe() function, nothing should be added to getPrompt()